core/controllers/ingest.py

```diff
@@ -313,3 +316,17 @@ def save_submission_file(excel_file, submission_id):
     submission.submission_file = excel_file.stream.read()
     db.session.add(submission)
     db.session.commit()
+
+
+def remove_unreferenced_organisations():
+    """Removes organisations no longer referenced by a Programme.
+
+    Some organisation names have been wrong in previous ingest.
+    This will clean them up everytime a new ingest is run updating those names
+    """
+
+    organisations_without_child = Organisation.query.filter(~Organisation.programmes.any()).all()
+    if organisations_without_child:
+        Organisation.query.filter(Organisation.id.not_in([org.id for org in organisations_without_child])).delete()
```

Comment:
does this work? It looks like you are trying to delete all the orgs Not without children, ie all the ones with children. Or am I getting confused here?